public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts

```diff
@@ -136,6 +136,21 @@ export async function promQailExplain(
     });
 }
 
+/**
+ * Check if the LLM plugin is enabled.
+ * @returns true if the LLM plugin is enabled.
+ */
+export async function isLLMPluginEnabled(): Promise<boolean> {
+  // Check if the LLM plugin is enabled.
+  // If not, we won't be able to make requests, so return early.
+  const openaiEnabled = llms.openai.enabled().then((response) => response.ok);
```

Comment:
I would consider this a workaround bit of code, so I'm thinking of when we fix @grafana-experimental so that `llms.openai.enabled()` reteurns a boolean as we originall planned. Something like this would be future proof, as it will decouple this plugin release from the fixed @grafana-experimental release.

```javascript
const openaiEnabled = llms.openai.enabled();
const vectorEnabled = llms.vector.enabled();
if (typeof openaiEnabled == "boolean") {
  return openaiEnabled && vectorEnabled;
}
// workaround for 1.7.4 @grafana-experimental bug
const openaiEnabledPromise = openaiEnabled.then((response) => response.ok);
const vectorEnabledPromise = vectorEnabled.then((response) => response.ok);
// combine 2 promises
return Promise.all([openaiEnabled, vectorEnabled]).then((results) => {
  return results.every((result) => result);
});
```

Reply:
This would work, but not sure if this has a big benefit and technically we're adding dead code that is never reached.
The version of grafana-experimental is pinned, so we don't need to worry about backward/forward compatibility in here. I'd say let's not worry about this here, and get things working on 1.7.4.
Once we release a new grafana-experimental 1.7.5, we'll need a separate PR anyway to bump it and update `yarn.lock`, and in that PR we can update the health check and tests to remove all of this.

Reply:
Ohh duh this is in grafana inself. I keep having a plugin mentality in mind. Sorry yes, my concerns are just wrong.